lib/rules/max-len.js

```diff
@@ -75,7 +75,6 @@ module.exports = {
         },
 
         schema: [
-            OPTIONS_OR_INTEGER_SCHEMA,
             OPTIONS_OR_INTEGER_SCHEMA,
             OPTIONS_SCHEMA
```

Comment:
This allows for configuration in the following format: [max_width, tab_length, options].

For example, this test case is failing after this change because the new schema does not allow [number, number, object]:

https://github.com/eslint/eslint/blob/bc692a9bf5c664c646ce386eff44eb706c231127/tests/lib/rules/max-len.js#L45-L48